src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -509,20 +535,26 @@ def deployment_options(self):
                 if device.available:
                     if device.human_readable_type == 'hdd':
                         hdds += 1
+                    # SSDs and NVMe are both counted as 'ssd'
+                    # so differentiating nvme using its path
+                    elif '/dev/nvme' in device.path:
+                        nvmes += 1
                     else:
                         ssds += 1
-                        # we still don't know how to infer nvmes
-                        # Tracker: https://tracker.ceph.com/issues/55728
-                        nvmes += 1
 
         if hdds:
             res.options[OsdDeploymentOptions.COST_CAPACITY].available = True
             res.recommended_option = OsdDeploymentOptions.COST_CAPACITY
-        if ssds:
+            res.options[OsdDeploymentOptions.COST_CAPACITY].hdd_used = hdds
+        if ssds and (hdds or nvmes):
```

Comment:
```suggestion
        if hdds and (ssds or nvmes)```
 I remember this being the througput case. Anyways we have to check if this is correct.

Reply:
hmm what I remember from the throughput option is that, throughput can be `hdds/ssds` where `hdds` is `data_devices` and `ssds` are `db_devices`. it can also bee `ssds/nvmes` where `ssds` can be `data_devices` and nvmes can be `db_devices`.